### DISCUSSION TITLE
Lesson 11 - how to make SVGs on-chain less expensive?

### DISCUSSION THREAD

**Discussion by sharonphiliplima** [Cyfrin/foundry-full-course-cu#581] [Q&A]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/69852010/2106bb3a-c64a-48fc-bd78-144fbe93497f)

I have an artificially generated artistic SVG code which is around 2.2MB! (47295 LOC) -> https://github.com/sharonphiliplima/MySVGs/blob/main/Girl.svg

Does it make good sense to make it decentralised and store it on chain? Why or why not?

A less expensive route is to store them in contract bytecode. 
If yes, how?

**Comment by theirrationalone** [comment]
First of all keep we want to sell NFT in mind thing... 

We should to make it decentralized by using services like ipfs. It should not be stored In a contract's bytecode... 

If somehow we were able to store it in bytecode then as a result it will be most Gas Expensive to us. It would also mean we made a most Expensive contract with static data into it. Only to us. 

Though this data will be stored as a string and with this much of data bytes32 can be reached that's also mean we could have a exhaustion issue. This might be a reason why we use ipfs like services. 


So we should pin 📍it onto data pinning services nodes like ipfs. And after selling nft it will cost once to us then the new holder will have its ownership.

After selling that NFT we're done with our business, then all the gas cost will be paid by its new owner... New owner -.> new owner -.> new owner -.> .......

An NFT should be Gas Affordable, sell able, purchasable. So now we have a reason to not to store it into bytecode. 

Sorry if i am wrong ❌ here... Everyone are Welcome to make me correct ✅.
